pydash/pydash_app/dashboard/aggregator.py

```diff
@@ -0,0 +1,63 @@
+import persistent
+
+
+class Aggregator(persistent.Persistent):
+    """
+    Maintains aggregate data for either a dashboard or a single endpoint.
+    This data is updated every time a new endpoint call is added.
+    """
+
+    def __init__(self, endpoint_calls):
+        """
+        Constructor
+        :param endpoint_calls: List of endpoint calls to keep track of
+        """
+
+        self._endpoint_calls = endpoint_calls
+
+        self.total_visits = 0
+        self.total_execution_time = 0
+
+        self.average_execution_time = None
+
+        self.visits_per_day = {}
+        self.visits_per_ip = {}
+
+    def add_endpoint_call(self, endpoint_call):
+        """
+        Add an endpoint call and update aggregated data
+        :param endpoint_call: `EndpointCall` instance to add
+        """
+
+        # precondition: the endpoint call has been added to the list of endpoint calls already
+
+        self.total_visits += 1
+        self.total_execution_time += endpoint_call.execution_time
```

Comment:
When taking data from endpoint calls, it might be better to use the get_data() method, in order to avoid using the internal representation of these values.

Reply:
In Python there's no real way to restrict access to attributes, and I'm not sure if getting the values from the get_data() method would be "pythonic". You're right though about not exposing the internals. Maybe @FlyingBird95 @Qqwy could chime in on this?